### ISSUE TITLE
New columns etc in template tables

### ISSUE THREAD

**Issue by amasses** [pgpartman/pg_partman#494]
With native partitioning, and template tables, I have a question around schema changes:

If we are making a schema change to the parent table (e.g. add a new column, or change the type of a column), I wanted to confirm that we also need to make this schema change for the template table as well - is this correct?

Thank you!

**Comment by keithf4** [comment]
That would be the best practice yes, so I would highly recommend just doing that.

But, technically, I believe that depends on what version of PostgreSQL you're using, what the change is and if it actually affects the things that are being inherited. Like if you simply added a new column, with no other changes, it wouldn't make a difference. But if you also wanted to add a new unique index using that new column, then that would be required.

The version of PG matters as well as older versions required the template table for more features (see the table in the docs https://github.com/pgpartman/pg_partman/blob/master/doc/pg_partman.md#child-table-property-inheritance). Most recent version is basically down to unique indexes and the unlogged property.

If you want to look more closely yourself, you can inspect the function that manages template inheritance - https://github.com/pgpartman/pg_partman/blob/master/sql/functions/inherit_template_properties.sql

**Comment by amasses** [comment]
@keithf4 thanks so much for confirming this. 

Just so I am completely clear, with PG14 the main purpose of the template table is to support a) `unlogged` options on tables, and b) support unique indices which do not include the partition column (but it is good practice to keep the schema changes aligned)?

**Comment by keithf4** [comment]
Yes, that is currently what it is there for for PG14 & 15. Could potentially be used for other things in the future if needed as well, hence the best practice of keeping the template in sync with the current parent.
